crates/proto/src/client.rs

```diff
@@ -0,0 +1,55 @@
+use std::time::Duration;
+
+use anyhow::Context;
+use tonic::{
+    metadata::AsciiMetadataValue,
+    service::{Interceptor, interceptor::InterceptedService},
+    transport::Channel,
+};
+use url::Url;
+
+use crate::generated::rpc::api_client::ApiClient;
+
+pub type RpcClient = ApiClient<InterceptedService<Channel, AcceptHeaderInterceptor>>;
```

Comment:
Should we change this into a proper newtype - i.e.,:
```rust
pub struct RpcClient(ApiClient<InterceptedService<Channel, AcceptHeaderInterceptor>>);
```
This way we could implement `connect()` like:
```rust
impl RpcClient {
    pub async fn connect(url: &Url, timeout_ms: u64) -> anyhow::Result<Self> {
        ...
    }
}
```

Reply:
I made this change and reworked the interceptor to be a bit more general so it makes sense for it to be used to inject other metadata if we ever need.

```rust
/// Alias for gRPC client that wraps the underlying client for the purposes of metadata configuration.
type InnerClient = ApiClient<InterceptedService<Channel, MetadataInterceptor>>;

/// Client for the Miden RPC API which is fully configured to communicate with a Miden node.
pub struct RpcClient(InnerClient);
```